pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -361,6 +415,12 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLis
 		})
 	}
 
+	if len(taskSpecSteps) > 0 {
```

Comment:
wondering if this check is really needed? :thinking:

Reply:
I change the code structure a bit in order to separate the sortTaskRun function.